File: ruby/difference-of-squares/difference_of_squares.rb
class Squares

  def initialize(num)
    @num = num
  end

  def square_of_sum
    nums = (1..@num).to_a
    sum = nums.reduce(:+)
    return sum ** 2
  end

  def sum_of_squares
    total = 0
    @num.times do | num |
      total += (num + 1)  ** 2
    end
    return total
  end

  def difference
    return square_of_sum - sum_of_squares
  end

end

module BookKeeping
  VERSION = 4
end

File: c/difference-of-squares/src/difference_of_squares.c
#include "difference_of_squares.h"
#include <stdio.h>

int sum_of_squares(int num)
{
  int total = 0 ;
  for ( int i = 1; i <= num; i++) {
    total +=  i * i;
  }
  return total;
}

int square_of_sum(int num)
{
  int total = 0 ;
  for ( int i = 1; i <= num; i++){
    total += i;
  }
  return total * total ;
}

int difference_of_squares(int num)
{
  return square_of_sum(num) - sum_of_squares(num);
}
